### PR TITLE
Needleman-Wunsch CUDA implementation

### PR DESCRIPTION
**Pull request by blackwer**
Tracking branch for a batch GPU CUDA Needleman-Wunsch implementation

### CODE REVIEW

**Comment by blackwer** [comment]
I don't have write permissions to your repo, so I couldn't perform the merge myself onto #27 . There were no substantial merge conflicts, so I went ahead and merged the tip of that PR here.

**Comment by mortonjt** [comment]
Got it - the previous implementation didn't easily implement batching, so the simulation code that I pushed in will need to be modified to handle this (something that I should tackle in a separate PR). As a result, the tests for that module aren't expected to pass.

It looks there are a few flake8 errors that are outstanding on nw_cuda.py (I can't fix those in an upstream patch). Otherwise, is this PR ready to be merged into master?

**Comment by blackwer** [comment]
I'm still implementing some optimizations over the weekend. There's a lot
that can be done, I think. Probably roughly a factor of two to be gained. I
can do this in a separate PR if you want to go ahead and merge this up

**Comment by mortonjt** [comment]
Got it.  Ok, let's merge this in - and open up another PR for further optimizations.

File: deepblast/nw_cuda.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+        Qd = torch.zeros((B, ZN, ZM, 3), dtype=Ztheta.dtype)
+        Vtd = torch.zeros(B, dtype=Ztheta.dtype)
+        Ed = torch.zeros((B, ZN, ZM), dtype=Ztheta.dtype)
+
+        d_Ztheta = cuda.to_device(Ztheta.detach().numpy())
+        d_ZA = cuda.to_device(ZA.detach().numpy())
+        d_Vtd = cuda.to_device(Vtd.detach().numpy())
+        d_Qd = cuda.to_device(Qd.detach().numpy())
+        d_Ed = cuda.to_device(Ed.detach().numpy())
```
Comment (mortonjt) [inline deepblast/nw_cuda.py:259]: Apparently in order to call numpy(), the data needs to be first be moved to the CPU.
I wonder if there is a way to bypass this (i.e. avoid calling numpy() in order to keep the data on the GPU).

**Pull request merged**